src/staticanalysis/bot/static_analysis_bot/workflow.py

```diff
@@ -149,25 +150,33 @@ def run(self, revision):
             # Only run mach if revision has any C/C++ or Java files
             if revision.has_clang_files:
 
-                # Mach pre-setup with mozconfig
-                try:
-                    logger.info('Mach configure...')
-                    with stats.api.timer('runtime.mach.configure'):
-                        run_check(['gecko-env', './mach', 'configure'], cwd=settings.repo_dir)
+                def do_build_setup():
```

Comment:
As this is a pretty hefty chunk of code, please move it in its own class function. That should make the main workflow more readable too.

Reply:
And move the ` with stats.api.timer('runtime.mach'):`  as a decorator of that function